File: ios/Sources/CapgoCapacitorDataStorageSqlitePlugin/Global.swift
//
//  Global.swift
//  Plugin
//
//  Created by  Quéau Jean Pierre on 08/04/2021.
//  Copyright © 2021 Max Lynch. All rights reserved.
//

import Foundation
struct Global {
    static var secretDefault: String = "test secret"
    static var newsecretDefault: String = "test new secret"

    var secret: String
    var newsecret: String

    init() {
        secret = Global.secretDefault
        newsecret = Global.newsecretDefault
    }

    static func configure(secret: String?, newSecret: String?) {
        if let secret = secret, !secret.isEmpty {
            secretDefault = secret
        }
        if let newSecret = newSecret, !newSecret.isEmpty {
            newsecretDefault = newSecret
        }
    }
}

File: android/src/main/java/com/jeep/plugin/capacitor/capgocapacitordatastoragesqlite/cdssUtils/Global.java
package com.jeep.plugin.capacitor.capgocapacitordatastoragesqlite.cdssUtils;

public class Global {

    private static final String DEFAULT_SECRET = "test secret";
    private static final String DEFAULT_NEW_SECRET = "test new secret";

    private static String secretDefault = DEFAULT_SECRET;
    private static String newSecretDefault = DEFAULT_NEW_SECRET;

    public String secret;
    public String newsecret;

    public Global() {
        secret = secretDefault;
        newsecret = newSecretDefault;
    }

    public static void configure(String secret, String newSecret) {
        if (secret != null && !secret.isEmpty()) {
            secretDefault = secret;
        }
        if (newSecret != null && !newSecret.isEmpty()) {
            newSecretDefault = newSecret;
        }
    }
}
